src/dawn/Compiler/DawnCompiler.cpp

```diff
@@ -158,69 +158,78 @@ std::unique_ptr<OptimizerContext> DawnCompiler::runOptimizer(std::shared_ptr<SIR
+    optimizer->checkAndPushBack<PassSSA>();
+    optimizer->checkAndPushBack<PassMultiStageSplitter>(mssSplitStrategy);
+    optimizer->checkAndPushBack<PassStageSplitter>();
+    optimizer->checkAndPushBack<PassPrintStencilGraph>();
+    optimizer->checkAndPushBack<PassTemporaryType>();
+    optimizer->checkAndPushBack<PassSetStageName>();
+    optimizer->checkAndPushBack<PassSetStageGraph>();
+    optimizer->checkAndPushBack<PassStageReordering>(reorderStrategy);
+    optimizer->checkAndPushBack<PassStageMerger>();
+    optimizer->checkAndPushBack<PassStencilSplitter>(maxFields);
+    optimizer->checkAndPushBack<PassTemporaryType>();
+    optimizer->checkAndPushBack<PassTemporaryMerger>();
+    optimizer->checkAndPushBack<PassInlining>(
+        (getOptions().InlineSF || getOptions().PassTmpToFunction),
+        PassInlining::InlineStrategy::ComputationsOnTheFly);
+    optimizer->checkAndPushBack<PassTemporaryToStencilFunction>();
+    optimizer->checkAndPushBack<PassSetNonTempCaches>();
+    optimizer->checkAndPushBack<PassSetCaches>();
+    optimizer->checkAndPushBack<PassComputeStageExtents>();
+    optimizer->checkAndPushBack<PassSetBoundaryCondition>();
+    optimizer->checkAndPushBack<PassSetBlockSize>();
+    optimizer->checkAndPushBack<PassDataLocalityMetric>();
+    optimizer->checkAndPushBack<PassSetSyncStage>();
+    // Since both cuda code generation as well as serialization do not support stencil-functions, we
+    // need to inline here as the last step
+    optimizer->checkAndPushBack<PassInlining>(getOptions().Backend == "cuda" ||
+                                                  getOptions().SerializeIIR,
+                                              PassInlining::InlineStrategy::ComputationsOnTheFly);
+    optimizer->checkAndPushBack<PassInlining>(true,
+                                              PassInlining::InlineStrategy::ComputationsOnTheFly);
+
+    DAWN_LOG(INFO) << "All the passes ran with the current command line arguments:";
+    for(const auto& a : optimizer->getPassManager().getPasses()) {
+      DAWN_LOG(INFO) << a->getName();
     }
-
-    stencil.second = instantiation;
+    int i = 0;
+    for(auto& stencil : optimizer->getStencilInstantiationMap()) {
+      // Run optimization passes
+      std::shared_ptr<iir::StencilInstantiation> instantiation = stencil.second;
+      DAWN_LOG(INFO) << "Starting Optimization and Analysis passes for `"
+                     << instantiation->getName() << "` ...";
+      if(!optimizer->getPassManager().runAllPassesOnStecilInstantiation(instantiation))
+        return nullptr;
+      DAWN_LOG(INFO) << "Done with Optimization and Analysis passes for `"
+                     << instantiation->getName() << "`";
+
+      if(options_->SerializeIIR) {
+        IIRSerializer::serialize(
+            remove_fileextension(instantiation->getMetaData().getFileName(), ".cpp") + "." +
+                std::to_string(i) + ".iir",
```

Comment:
The numbering here is because there could be multiple stencils in the same file, right?

Reply:
yes